src/components/instructor/AssignmentsView.js

```diff
@@ -10,15 +17,132 @@ import {useLocation} from 'react-router-dom'
 // assignment id, title, dueDate and buttons to grade, edit, delete each assignment
 
 const AssignmentsView = (props) => {
+    const headers = ['Id', 'Title', 'Due Date', 'Course Id', 'Sec Id', 'Sec No', '', ''];
+    const [asgnmts, setAsgnmts] = useState([
+
+    ]);
+    const [message, setMessage] = useState('');
+    const [search, setSearch] = useState({id: '', title: '', dueDate: '', secId: '', secNo: ''});
 
     const location = useLocation();
-    const {secNo, courseId, secId} = location.state;
-     
-    return(
-        <> 
-           <h3>Not implemented</h3>
+    const { secNo, courseId, secId } = location.state;
+
+    const fetchAsgnmts = async () => {
+        // TODO: get the assignments
+        try{
+            const response = await fetch(`${SERVER_URL}/sections/${secNo}/assignments`);
+            if (response.ok){
+                const asgnmts = await response.json();
+                setAsgnmts(asgnmts);
+            } else {
+                const json = await response.json();
+                setMessage("response error: "+ json.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+ err);
+        }
+    }
+
+    useEffect( () => {
+        fetchAsgnmts();
+    }, []);
+
+    const gradeAsgnmt = async (asgnmt) => {
+        try {
+            const response = await fetch(`${SERVER_URL}/grades`,
+                {
+                    method: 'PUT',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(asgnmt),
+                });
+            if (response.ok) {
+                setMessage("Grade Saved")
+                fetchAsgnmts();
+            } else {
+                const rc = await response.json();
+                setMessage(rc.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+err);
+        }
+    }
+
+    const editChange = (event) => {
+        setSearch({...search,  [event.target.name]:event.target.value});
+    }
+    const deleteAsgnmt = async (id) => {
+        try {
+            const response = await fetch (`${SERVER_URL}/assignments/${id}`,
+                {
+                    method: 'DELETE',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                });
+            if (response.ok) {
+                setMessage("Assignment deleted");
+                fetchAsgnmts();
+            } else {
+                const rc = await response.json();
+                setMessage(rc.message);
+            }
+        } catch (err) {
+            setMessage("network error: "+err);
+        }
+    }
+
+    const onDelete = (e) => {
+        const row_idx = e.target.parentNode.parentNode.rowIndex - 1;
+        const id = asgnmts[row_idx].id;
+        confirmAlert({
```

Comment:
Maybe add logic here to check if the assignment has a grade/ grade is not null, and then display an error that the assignment cannot be deleted because it has been graded.. or something like that.

Reply:
Looks like the fix @JADuarte18 made to backend resolves this issue.